ambry-frontend/src/main/java/com/github/ambry/frontend/SimpleUserQuotaRequestCostPolicy.java

```diff
@@ -0,0 +1,112 @@
+
+import com.github.ambry.messageformat.BlobInfo;
+import com.github.ambry.quota.QuotaName;
+import com.github.ambry.quota.RequestCostPolicy;
+import com.github.ambry.rest.RestMethod;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.rest.RestResponseChannel;
+import com.github.ambry.rest.RestUtils;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * {@link RequestCostPolicy} implementation that calculates request cost in terms of metrics tracked by user quota - capacity unit and storage.
+ * Capacity unit cost is defined as the number of 4MB chunks. Storage cost is defined as number of GB of storage used.
+ */
+public class SimpleUserQuotaRequestCostPolicy implements RequestCostPolicy {
+  final static double CU_COST_UNIT = 4 * 1024 * 1024; //4 MB
+  final static double BYTES_IN_GB = 1024 * 1024 * 1024;
+  final static double DEFAULT_COST_FOR_HEAD_DELETE_TTL = 1;
+
+  @Override
+  public Map<String, Double> calculateRequestCost(RestRequest restRequest, RestResponseChannel restResponseChannel,
+      BlobInfo blobInfo) {
+    Map<String, Double> costMap = new HashMap<>();
+    if (blobInfo != null) {
+      costMap.put(restMethodToCostMetric(restRequest.getRestMethod()),
+          calculateCapacityUnitCost(restRequest, blobInfo));
+      costMap.put(QuotaName.STORAGE_IN_GB.name(), calculateStorageCost(restRequest, restResponseChannel, blobInfo));
+    }
+    return costMap;
+  }
+
+  /**
+   * Get appropriate cost metric name for {@link RestMethod}.
+   * @param restMethod {@link RestMethod} object.
+   * @return CUQuotaMethod name for specified {@link RestMethod}.
+   */
+  private String restMethodToCostMetric(RestMethod restMethod) {
+    switch (restMethod) {
+      case GET:
+      case OPTIONS:
+      case HEAD:
+        return QuotaName.READ_CAPACITY_UNIT.name();
+      default:
+        return QuotaName.WRITE_CAPACITY_UNIT.name();
+    }
+  }
+
+  /**
+   * Calculate the storage cost incurred to serve a request.
+   * For post requests this is the number bytes in GB of the blob data. For all other write requests the default is DEFAULT_COST_FOR_HEAD_DELETE_TTL.
+   * For read requests there is no storage cost incurred.
+   * @param restRequest {@link RestRequest} to find type of request.
+   * @param restResponseChannel {@link RestResponseChannel} object.
+   * @param blobInfo {@link BlobInfo} with information about the blob.
+   * @return storage cost.
+   */
+  private double calculateStorageCost(RestRequest restRequest, RestResponseChannel restResponseChannel,
+      BlobInfo blobInfo) {
+    switch (restRequest.getRestMethod()) {
+      case POST:
+        long contentSize = 0;
+        if (restRequest.getUri().contains(Operations.STITCH)) {
+          contentSize = Long.parseLong((String) restResponseChannel.getHeader(RestUtils.Headers.BLOB_SIZE));
+        } else {
+          contentSize = restRequest.getBytesReceived();
+        }
+        return contentSize / BYTES_IN_GB;
```

Comment:
Does it makes sense to track storage cost at the request (as opposed to container) level in GB?  For a 10K blob, this value will be 0.000001.